### PR TITLE
kick off NIO2

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

NIO2 development starts now.

Modifications:

Made NIO Swift 5-only for everything else see docs/public-api-changes-NIO1-to-NIO2.md

Result:

NIO2 development can start.

### CODE REVIEW

File: docs/public-api-changes-NIO1-to-NIO2.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Changes in the Public API from NIO 1 to NIO 2
+
+- removed all previously deprecated functions, types and modules.
```
Comment (Lukasa) [inline docs/public-api-changes-NIO1-to-NIO2.md]: Do we need to have an explicit list of removals?

File: docs/public-api-changes-NIO1-to-NIO2.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+- made `EventLoopGroup.makeIterator()` a required method
+- `TimeAmount.Value` is not `Int64` (from `Int` on 64 bit platforms, no change
```
Comment (Lukasa) [inline docs/public-api-changes-NIO1-to-NIO2.md]: s/not/now/

File: docs/public-api-changes-NIO1-to-NIO2.md (hunk 1 above)
Comment (weissi) [reply docs/public-api-changes-NIO1-to-NIO2.md]: @Lukasa I don't think we need this for stuff that we had previously deprecated. I don't think we need to have a good migration story for people that use deprecated API.

**Comment by tomerd** [comment]
@swift-nio-bot test this please

**Comment by tomerd** [comment]
@swift-nio-bot test this please

**Comment by tomerd** [comment]
@swift-nio-bot test this please

**Comment by weissi** [comment]
@tomerd / @Lukasa this will unblock this other PRs until  the https://bugs.swift.org/browse/SR-9204 fix lands in the swift 5 dev previews

**Pull request merged**